packages/integration-karma/test/component/LightningElement.errorCallback/index.spec.js

```diff
@@ -1,153 +1,134 @@
 import { createElement } from 'lwc';
-import XErrorBoundary from 'x/errorBoundary';
+
+import XBoundaryChildConstructorThrow from 'x/boundaryChildConstructorThrow';
+import XBoundaryChildConnectedThrow from 'x/boundaryChildConnectedThrow';
+import XBoundaryChildRenderThrow from 'x/boundaryChildRenderThrow';
+import XBoundaryChildRenderedThrow from 'x/boundaryChildRenderedThrow';
+import XBoundaryChildSlotThrow from 'x/boundaryChildSlotThrow';
+import XNestedBoundaryChildThrow from 'x/nestedBoundaryChildThrow';
+import XBoundaryChildSelfRehydrateThrow from 'x/boundaryChildSelfRehydrateThrow';
+import XBoundaryAlternativeViewThrow from 'x/boundaryAlternativeViewThrow';
+
 import XChildConstructorThrowDuringInit from 'x/childConstructorThrowDuringInit';
 import XChildRenderThrowDuringInit from 'x/childRenderThrowDuringInit';
 import XChildRenderedThrowDuringInit from 'x/childRenderedThrowDuringInit';
 import XChildConnectedThrowDuringInit from 'x/childConnectedThrowDuringInit';
 
-// Wait for a macrotask because the test has to wait for a nested rehydration(async) to complete.
-// tl;dr
-// The tests are setup in a way where, each test component is enabled by changing a
-// tracked property on the container('x-error-boundary-suite'). This triggers an rehydration.
-// Next, there is an intermediate component that has errorCallback() to handle errors in the nested
-// child's life cycle. The intermediate component triggers a self rehydration by changing its state
-// in the errorCallback()
-function waitForNestedRehydration() {
-    return new Promise(resolve => {
-        setTimeout(resolve);
-    });
-}
-
 describe('error boundary', () => {
-    let elm;
-    let shadowRoot;
-    beforeEach(() => {
-        elm = createElement('x-error-boundary-suite', { is: XErrorBoundary });
+    it('should render alternative view if child throws in renderedCallback()', () => {
+        const elm = createElement('x-boundary-child-rendered-throw', {
+            is: XBoundaryChildRenderedThrow,
+        });
         document.body.appendChild(elm);
-        shadowRoot = elm.shadowRoot;
-    });
 
-    it('should render alternative view if child throws in renderedCallback()', () => {
-        elm.toggleFlag('boundaryChildRenderedThrow');
-        return waitForNestedRehydration().then(() => {
-            const innerShadowRoot = shadowRoot.querySelector('x-boundary-child-rendered-throw')
-                .shadowRoot;
-            const altenativeView = innerShadowRoot.querySelector('.rendered-calback-altenative');
-            expect(altenativeView.textContent).toEqual('renderedCallback alternative view');
+        return Promise.resolve().then(() => {
```

Comment:
we didn't have this before, but would adding expect.assertions add more certainty to the test?

Reply:
No need for this, the assertion count is only needed when running an async test where the assertion code is not related to the code triggering the end of the test. Let's take examples to illustrate this:

```js
test('should invoke foo', () => {
  doSomethingSync(data => {
    expect(data).not.toBeUndefined();
  });
})
```

In the test above we need to make sure the assertion runs because the test will pass even if the callback is not invoked.

```js
test('should invoke foo', done => {
  doSomethingAsync(data => {
    expect(data).not.toBeUndefined();
    done();
  });
});
```

In the test above, the assertion count is not necessary. For the test to finish its execution it has to run into the callback and has to run the assertion. The same observation also applies to tests returning promises.